src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -798,8 +798,8 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
                     break;
             }
 
-            Checks.check(!(activityType != MessageActivity.ActivityType.LISTENING && application == null), "Either the ActivityType is wrong or the Application is null!");
-            activity = new MessageActivity(activityType, partyId, application);
+            if (!(activityType != MessageActivity.ActivityType.LISTENING && application == null))
```

Comment:
This is not really how you're supposed to invert boolean expressions, you can just flip operators with de-morgan

Reply:
bump

Reply:
I dont know if my old data is still valid so i removed the if statement.